build_tools/compiler/template_processor.rb

```diff
@@ -84,8 +85,10 @@ def extractable_options?(options)
 
     def stringify_keys(options)
       result = {}
-      options.each_key do |key|
-        result[key.to_s] = options[key]
+      if !options.empty?
```

Comment:
minor nitpick: do we need this conditional?  if options is `.empty?` then the `.each_key` iterator won't do anything, so it should be safe to leave this method as it was.